BDArmory/Utils/ProjectileUtils.cs

```diff
@@ -9,19 +9,70 @@
 using BDArmory.FX;
 using BDArmory.GameModes;
 using BDArmory.Settings;
+using System.IO;
 
 namespace BDArmory.Utils
 {
     class ProjectileUtils
     {
-        static HashSet<string> IgnoredPartNames;
+        public static string settingsConfigURL = Path.Combine(KSPUtil.ApplicationRootPath, "GameData/BDArmory/PluginData/PartsBlacklists.cfg");
+        public static void SetUpPartsHashSets()
+        {
+            var fileNode = ConfigNode.Load(settingsConfigURL);
+            if (fileNode == null)
+            {
+                fileNode = new ConfigNode();
+                if (!Directory.GetParent(settingsConfigURL).Exists)
+                { Directory.GetParent(settingsConfigURL).Create(); }
+                if (!fileNode.HasNode("IgnoredParts"))
+                {
+                    fileNode.AddNode("IgnoredParts");
+                }
+                ConfigNode Iparts = fileNode.GetNode("IgnoredParts");
+                Iparts.SetValue("Part1", "ladder1", true);
+                Iparts.SetValue("Part2", "telescopicLadder", true);
+                Iparts.SetValue("Part3", "telescopicLadderBay", true);
+
+                if (!fileNode.HasNode("MaterialsBlacklist"))
+                {
+                    fileNode.AddNode("MaterialsBlacklist");
+                }
+                ConfigNode BLparts = fileNode.GetNode("MaterialsBlacklist");
+                BLparts.SetValue("Part1", "InflatableHeatShield", true);
+                BLparts.SetValue("Part2", "foldingRad*", true);
+                BLparts.SetValue("Part3", "radPanel*", true);
+                BLparts.SetValue("Part4", "ISRU*", true);
+                BLparts.SetValue("Part5", "Scanner*", true);
+                BLparts.SetValue("Part5", "Drill*", true);
+
+                fileNode.Save(settingsConfigURL);
+            }
+        }
+            static HashSet<string> IgnoredPartNames;
         public static bool IsIgnoredPart(Part part)
         {
             if (IgnoredPartNames == null)
             {
                 IgnoredPartNames = new HashSet<string> { "bdPilotAI", "bdShipAI", "missileController", "bdammGuidanceModule" };
                 IgnoredPartNames.UnionWith(PartLoader.LoadedPartsList.Select(p => p.partPrefab.partInfo.name).Where(name => name.Contains("flagPart")));
                 IgnoredPartNames.UnionWith(PartLoader.LoadedPartsList.Select(p => p.partPrefab.partInfo.name).Where(name => name.Contains("conformaldecals")));
+
```

Comment:
Probably should transfer over flags to the .cfg on the off-chance someone wants them back as cheap spaced armor or something